test_build_tag_push.py

```diff
@@ -0,0 +1,99 @@
+import unittest
+import build_tag_push
+import os
+import subprocess
+import argparse
+from unittest.mock import patch
+import io
+import sys
+
+path_to_test_yml = ('/Users/alyssakelley/Desktop/MVPStudio/build_tag_push/'
+                    'test_data/builds_succeeds_example/app.yml')
+
+path_to_fail_test_yml = ('/Users/alyssakelley/Desktop/MVPStudio/'
+                         'build_tag_push/test_data/build_fails_example/'
+                         'app.yml')
+
+
+class TestBuildTagPush(unittest.TestCase):
+
+    def test_parse_app_file(self):
+
+        data_from_app_yml = build_tag_push.parse_app_file(path_to_test_yml)
+        self.assertEqual(data_from_app_yml['name'], 'build-setup-test')
+        self.assertEqual(data_from_app_yml['build'], 'build.sh')
+        self.assertEqual(data_from_app_yml['test'], 'test.sh')
+        self.assertEqual(data_from_app_yml['docker_context'],
+                         ['build', 'static', 'Dockerfile'])
+
+    def test_build_check_app_throws_on_build(self):
+        data_from_app_yml = build_tag_push.parse_app_file(path_to_test_yml)
+
+        with self.assertRaises(
+             build_tag_push.BuildFailedException) as shouldThrow:
+
+            build_tag_push.build_check_app(data_from_app_yml,
+                                           path_to_fail_test_yml, True)
+
+        self.assertEqual("The build failed.", shouldThrow.exception.message)
+
+    def test_build_check_app_runs_build(self):
+        data_from_app_yml = build_tag_push.parse_app_file(path_to_test_yml)
+        build_tag_push.build_check_app(data_from_app_yml, path_to_test_yml,
+                                       True)
+        self.assertTrue(os.path.exists('/Users/alyssakelley/Desktop/MVPStudio/'
```

Comment:
There's a few issues here:

* The paths are hard coded so they only work on your machine
* The test leaves an output file on your disk which will affect the next run of the test. For example, imagine you ran the tests and they passed, then you changed your code in a way that broke it: the test here would still pass because the file is still there from a previous run.

I think maybe the easiest thing to do is have this script delete the file when it starts - then run the build and then you can check that it was re-created.